include/simdjson/generic/ondemand/value_iterator-inl.h

```diff
@@ -142,6 +142,9 @@ simdjson_warn_unused simdjson_really_inline simdjson_result<bool> value_iterator
 simdjson_warn_unused simdjson_really_inline simdjson_result<bool> value_iterator::find_field_unordered_raw(const std::string_view key) noexcept {
   error_code error;
   bool has_value;
+  // We want to be able to loop back to where we were.
```

Comment:
I'm trying to figure out how this change affects things. Before:

1. In the initial if() statement, we either move to the beginning of a field, or set `has_value = false`. `search_start` is set here.
2. We scan fields until we either find (and return) a match, or `has_value == false`.
3. Repeat steps 1-2 from after resetting just past the opening `{`, which is either the beginning of a field or the terminating `}` (in which case `started_value()` sets `has_value = false`).

The new code sets `search_start` to the earlier position, which could be at the start of the object after the `{`, or in the middle of a field value, or at a `,`, or at the terminating `}'. This necessitates the extra `if()` at the end, but I'm not sure what it fixes?

Reply:
(I'm not denying it fixes the problem btw, it just feels like there must be a bug in the initial algorithm we still haven't fixed, and we may be papering over it here.)

Reply:
See my notes. I believe that the keys are at the same depth as the object. You only go down a level (increment the depth) when you enter a value. When you search for a key that is found, all is good because you exit at the key level. However, when the key is not found, look at what happens at the end... you access the colon, and increment the depth. So the depth you end up with depends on whether the key is found or not. What my patch does is make sure that if we started looking at a key, we come back at that key, and not a little bit further (and with depth that is incremented).